app/controllers/companies_controller.rb

```diff
@@ -0,0 +1,54 @@
+class CompaniesController < ApplicationController
+
+  before_action :authenticate_user!
+  before_action :find_company_by_id,  except: [:new, :create, :index]
+
+  def new
+    @company = Company.new
+  end
+
+  def index
+    @companies = Company.all
+  end
+
+  def create
+    @company = Company.new(company_params)
+    if @company.save
+      flash[:success] = t('.success')
+    else
+      flash[:error] = t('.error')
```

Comment:
@AmiFaldu-BTC Please do not redirect in case of failure, We need to show the page again with the filled values. Use render instead of the redirect in case of failure and explain the same to other as well.